inbm/dispatcher-agent/dispatcher/fota/bios_factory.py

```diff
@@ -195,6 +194,20 @@ def __init__(self,
                  params: Dict) -> None:
         super().__init__(dispatcher_broker, repo, params)
 
+    def _get_guid_from_system(self, manifest_guid: Optional[str]) -> str:
+        # get the GUID from the system using FW tool
+        extracted_guids = extract_guids(
+            self._fw_tool, ["System Firmware type", "system-firmware type"])
+        if manifest_guid:
+            if manifest_guid not in extracted_guids:
+                raise FotaError(
+                    f"GUID in manifest does not match any system firmware GUID on the system")
+            else:
+                return manifest_guid
+        else:
+            return extracted_guids[0]
```

Comment:
**<h2>$\color{#FFD700}Coding Style$</h2>**
**Suggestion:** To enhance code readability and maintainability, consider refactoring the conditional logic by using guard clauses.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if not manifest_guid:
            return extracted_guids[0]
        if manifest_guid not in extracted_guids:
            raise FotaError(f"GUID in manifest does not match any system firmware GUID on the system")
        return manifest_guid
```